File: src/Medium/Leetcode74.java
package Medium;

//Referred LC approach 1
public class Leetcode74 {
    public boolean searchMatrix(int[][] matrix, int target) {
        int left=0;
        int right=matrix[0].length*matrix.length-1;
        while(left<=right){
            int mid=left+(right-left)/2;
            if(target==matrix[mid/matrix[0].length][mid%matrix[0].length]){
                return true;
            }
            if(target>matrix[mid/matrix[0].length][mid%matrix[0].length]){
                left=mid+1;
            }
            else{
                right=mid-1;
            }
        }
        return false;
    }
}

File: src/Python/Medium/Leetcode74.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        left=0
        right=len(matrix)*len(matrix[0])-1
        while left<=right:
            mid=(left+right)//2
            midRow=mid//len(matrix[0])
            midCol=mid%len(matrix[0])

            if matrix[midRow][midCol]==target:
                return True
            elif matrix[midRow][midCol]>target:
                right=mid-1
            else:
                left=mid+1

        return False